src/software/jetson_nano/services/power.cpp

```diff
@@ -0,0 +1,57 @@
+#include "software/jetson_nano/services/power.h"
+
+#include <cstdint>
+
+#include "shared/uart_framing/uart_framing.hpp"
+
+PowerService::PowerService()
+    : status(), READ_BUFFER_SIZE(getMarshalledSize(PowerStatus()))
+{
+    this->uart = std::make_unique<BoostUartCommunication>(io_service, BAUD_RATE,
+                                                          DEVICE_SERIAL_PORT);
+}
+
+void PowerService::start() {}
+
+void PowerService::stop() {}
+
+std::unique_ptr<PowerStatus> PowerService::poll(const PowerCommand& command)
+{
+    auto frame                = createUartMessageFrame(command);
+    auto power_command_buffer = frame.marshallUartPacket();
+
+    std::vector<uint8_t> power_status_msg;
+    try
+    {
+        // Write power command
+        if (uart->serialWrite(power_command_buffer))
+        {
+            uart->flushSerialPort(uart->flush_send);
+        }
+        else
+        {
+            LOG(WARNING) << "Writing power command failed.";
+        }
+
+        // Read power status
+        uart->flushSerialPort(uart->flush_receive);
+        power_status_msg = uart->serialRead(READ_BUFFER_SIZE);
+    }
+    catch (std::exception& e)
+    {
+        LOG(FATAL) << "ESP32 has disconnected. Power service has crashed" << e.what();
```

Comment:
we _should_ technically be able to run without kick/chip, but its not clear what we would do here rn

eventually this should be an error code, but we can revisit it